tests/test-books-api.js

```diff
@@ -0,0 +1,178 @@
+            const book_name = 'Naruto'
+            const book_author = 'Masashi Kishimoto'
+            const book_publisher = 'Kim Dong'
+            const book_quantity = 230
+            const book_id_category = '1c2ad358-98b0-4210-9ea3-3c4625b76e91'
+            let response = await axios.put(`${process.env.API_URL}/books/${book_id}`, {
+                isbn: book_isbn,
+                name: book_name,
+                author: book_author,
+                publisher: book_publisher,
+                quantity: book_quantity,
+                id_category: book_id_category
+            })
+            assert.strictEqual(response.status, 200)
+            assert.strictEqual(response.data.State, State.SUCCESS)
+
+            let result = await dbbooks.get_book_by_id_async(book_id)
+            assert.strictEqual(result.rows[0].isbn, book_isbn)
+            assert.strictEqual(result.rows[0].name, book_name)
+            assert.strictEqual(result.rows[0].author, book_author)
+            assert.strictEqual(result.rows[0].publisher, book_publisher)
+            assert.strictEqual(result.rows[0].quantity, book_quantity)
+            assert.strictEqual(result.rows[0].id_category, book_id_category)
+        })
+
+        it('8.should return status 500 and the State response will be error when called with no parameters', async() => {
+            let response = await axios.put(`${process.env.API_URL}/books/${book_id}`, {}).catch(error => {
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+            })
+        })
+    })
+    
+    describe('delete a book', async() => {
+        it('9.should return status 200 and delete information of book in the database when called', async() => {
+            let response = await axios.delete(`${process.env.API_URL}/books/${book_id}`)
+            let result = await dbbooks.get_book_by_id_async(book_id)
+            assert.strictEqual(response.status, 200)
+            assert.strictEqual(response.data.DeletedBookId, book_id)
+            assert.strictEqual(response.data.State, State.SUCCESS)
+            assert.strictEqual(result.rows.length, 0)
+        })
+
+        it('10. should return status 500 and the State response will be error when called with wrong id\'s format', async() => {
+            let response = await axios.delete(`${process.env.API_URL}/books/1`).catch(error => {
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+        })
+
+        it('11. should return status 500 and the State response will be error when called with id\'s value', async() => {
+            let new_id = null
+            if(book_id.charAt(35) != 1) {
```

Comment:
There is an easier way to change Id, but can we 100% sure the changed Id hasn't already in the database?

Reply:
I fixed it